src/protocol.rs

```diff
@@ -313,42 +324,60 @@ impl std::fmt::Display for ParticipantIdentifier {
-/// created from sid and other post key generation shared parameters for the
-/// purpose of pre/signing. The codebase does not make this kind of distinction
-/// relying instead on a single type for all session identifiers. The
-/// distinction between sessions and sub-sessions is inherently enforced by the
-/// order of inputs and outputs to different stages in the protocol.
+/// 2. The paper distinguishes between sessions (keygen) and sub-sessions
+/// (auxinfo and presign).
+/// The codebase requires the calling application to select a new, unique
+/// session [`Identifier`]s at three points:
+/// (1) immediately before starting a new keygen session;
+/// (2) immediately before starting a new auxinfo session;
+/// (3) immediately before starting a new presigning session.
 ///
-/// Discrepancy (B): The codebase  instantiates [`Identifier`]s in three
-/// different ways: (1) as a session identifier for keygen, that are created at
-/// the start of a key generation instance; (2) as a session identifier for
-/// pre-signing, (3) as a session identifier for auxiliary information
-/// generation sessions; The paper itself only distinguishes between sessions
-/// and sub-sessions and combine (2) and (3).
+///     In fact, we think this is a minor bug in Figure 6 of the paper, since
+/// the definition of `ssid` includes outputs of auxinfo, and thus cannot be
+/// passed as input to auxinfo. We believe the correct instantiation of the
+/// `ssid` for auxinfo is in Figure 3, which includes fixed parameters and
+/// outputs from keygen, but _not_ outputs from auxinfo.
 ///
-/// Discrepancy (C): In the paper ssid is periodically refreshed with the
-/// session key and auxiliary information. The codebase does not do that and
-/// instead relies on the calling application to generate and refresh these
-/// identifiers by randomly sampling a new and unique 32 bytes identifier using
-/// `Identifier::random()`. This assumes that the participants initiating a
-/// protocol run are honestly generating globally unique identifiers and
-/// distributing them to the correct set of parties.
+/// 3. 🔒 In the paper, `ssid` is updated each time the participants run the
+/// key-refresh subprotocol.
+/// The codebase relies on the calling application to generate a new, unique
+/// `Identifier` for each new session.
 ///
-/// TODO: Discrepancy (C) needs to be further addressed by issue #218.
+///     This requires the calling application to select a protocol with
+/// appropriate trust assumptions for its deployment to ensure the resulting
+/// [`Identifier`] is unique and that all parties have the same one.
+/// Sample protocols (with varying trust models!) could include:
+///     - A trusted party randomly samples a unique identifier with
+///       `Identifier::random()` and sends it to all parties;
+///     - The participants run a Byzantine agreement protocol;
+///     - The participants use a pseudorandom correlation generator.
```

Comment:
ah, here it is. I think we should move this up top, to where we first start talking about session identifiers.

Reply:
You don't need a PCG. It's just an option. 

Say you have parties P1, ..., Pn. They then want to generate 10 ECDSA keys. Can you explain how following the paper's suggestion for `sid` results in a unique session identifier for each of these 10 separate calls to KeyGen? I see only static information included.

Reply:
To add more on the PCG option: it's definitely overkill, but it's one way for parties to agree on values without introducing extra rounds during protocol runs. You eat the setup cost up front.

Reply:
Moved!

Reply:
What I still don't understand is what the correlation you are hoping to get out of a PCG is ? Wouldn't you get your desired outcome if the parties use a PRG with the same seed ?

Reply:
Sure - but isn't that technically still a PCG? 

I just had PCG techniques in my head when we wrote this list, because I was thinking about them for another problem. Would you prefer we remove this reference altogether? Reword in some fashion? I don't have any strong feelings about keeping/removing.

Reply:
It is a PCG, but that seems like a very big hammer for what amounts to a PRG. Unless I am missing something. 
I'd rather if we removed this reference altogether.

Reply:
Comically oversized hammer is correct, I'll take it out.